as.c

```diff
@@ -304,18 +373,46 @@ DownloadProfile(HANDLE hWnd, const WCHAR *host, const char *username, const char
     }
 
     /* get http status code */
-    DWORD statusCode = 0;
+    DWORD status_code = 0;
     DWORD length = sizeof(DWORD);
-    HttpQueryInfoW(hRequest, HTTP_QUERY_STATUS_CODE | HTTP_QUERY_FLAG_NUMBER, &statusCode, &length, NULL);
-    if (statusCode != 200) {
-        ShowLocalizedMsgEx(MB_OK, hWnd, _T(PACKAGE_NAME), IDS_ERR_URL_IMPORT_PROFILE, statusCode, "HTTP error");
-        goto done;
-    }
+    HttpQueryInfoW(hRequest, HTTP_QUERY_STATUS_CODE | HTTP_QUERY_FLAG_NUMBER, &status_code, &length, NULL);
 
-    /* download profile content */
     size_t size = 0;
-    if (!DownloadProfileContent(hWnd, hRequest, &buf, &size))
+
+    /* download profile content */
+    if ((status_code == 200) || (status_code == 401)) {
+        if (!DownloadProfileContent(hWnd, hRequest, &buf, &size))
```

Comment:
Would the user get any feedback if this happens? Show an error?

Reply:
`DownloadProfileContent` displays an error message in case of failure. I added this to doxygen comment.